tests/s25Main/UI/testMapSelection.cpp

```diff
@@ -0,0 +1,246 @@
+// Copyright (C) 2005 - 2021 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Loader.h"
+#include "RttrForeachPt.h"
+#include "controls/ctrlMapSelection.h"
+#include "driver/MouseCoords.h"
+#include "uiHelper/uiHelpers.hpp"
+#include "rttr/test/TmpFolder.hpp"
+#include <libsiedler2/Archiv.h>
+#include <libsiedler2/ArchivItem_Bitmap_Raw.h>
+#include <libsiedler2/PixelBufferBGRA.h>
+#include <libsiedler2/libsiedler2.h>
+#include <rttr/test/random.hpp>
+#include <turtle/mock.hpp>
+
+using namespace rttr::test;
+
+namespace {
+MOCK_BASE_CLASS(TestWindow, Window)
+{public: TestWindow(): Window(nullptr, randomValue<unsigned>(), DrawPoint(0, 0)){}};
+
+std::unique_ptr<libsiedler2::Archiv> createBitmapFromBuffer(Extent const& size,
+                                                            libsiedler2::PixelBufferBGRA const& buffer,
+                                                            const libsiedler2::ArchivItem_Palette* palette = nullptr)
+{
+    using namespace libsiedler2;
+    auto bmpRaw = std::make_unique<ArchivItem_Bitmap_Raw>();
+    bmpRaw->init(size.x, size.y, TextureFormat::BGRA);
+    bmpRaw->create(buffer, palette);
```

Comment:
You can combine both when using a PixelBuffer:
```suggestion
    bmpRaw->create(buffer, palette);
```